File: 0069-sqrtx/0069-sqrtx.java
class Solution {
    public int mySqrt(int x) {
        if (x==0) return 0;
        int l = 1, r = x;
        while(l<=r){
            int m = l + (r-l)/2;
            if(m == x/m) return m;
            if (m<x/m) l = m+1 ;
            else r = m-1;
        }
        return r;
    }
}

File: 0069-sqrtx/0069-sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if x == 0 : return 0
        r, l = x, 0
        while(l<=r):
            m = l + (r-l)//2
            if(m * m == x): return m
            if(m * m < x): l = m+1
            else : r = m-1
        return r
        